tests/MMS/spatial/fci/mms.py

```diff
@@ -26,22 +25,11 @@
 Bpx = Bp + (x-0.5)*Lx * Bpprime  # Note: x in range [0,1]
 B = sqrt(Bpx**2 + Bt**2)
 
-def FCI_Grad_par(f):
-    return ( Bt * diff(f, y)*2.*pi/Ly + Bpx * diff(f, z)*2.*pi/Lz ) / B
+def FCI_ddy(f):
+    return ( Bt * diff(f, y)*2.*pi/Ly + Bpx * diff(f, z)*2.*pi/Lz )
```

Comment:
Not sure about this: Dividing by B normalises the Bt and Bpx components, otherwise the result depends on the magnitude of B, not just the direction.

Reply:
This was originally because I'd changed the MMS test to `DDY`, but it turns out that doesn't really make much sense for FCI, and `Grad_par` should always be used instead. `Grad_par` does require this `B` here and also in `g_22`

I've done the easy thing and added a simple calculation of the field line length to zoidberg, but it could really do with doing it properly. This is a bit tricky without breaking lots of things!